File: contracts/get_abi.py
import requests
import json

url = "https://api.arbiscan.io/api?module=contract&action=getabi&address=0xc873fEcbd354f5A56E00E710B90EF4201db2448d"
response = requests.get(url)
data = response.json()
if data['status'] == '1':
    abi = json.loads(data['result'])
    funcs = [f for f in abi if f.get('type') == 'function' and f.get('name') == 'exactInputSingle']
    print(json.dumps(funcs, indent=2))
else:
    print("Error:", data)

File: contracts/get_abi.js
const https = require('https');
https.get("https://api.arbiscan.io/api?module=contract&action=getabi&address=0xc873fEcbd354f5A56E00E710B90EF4201db2448d", res => {
    let data = '';
    res.on('data', c => data += c);
    res.on('end', () => {
        const json = JSON.parse(data);
        if (json.status === "1") {
            const abi = JSON.parse(json.result);
            const func = abi.find(a => a.name === "exactInputSingle");
            console.log(JSON.stringify(func, null, 2));
        } else {
            console.log("Error:", json);
        }
    });
});
